src/scripts/push_coral_to_hub.py

```diff
@@ -0,0 +1,181 @@
+"""Pushes the processed CoRal dataset to the Hugging Face Hub.
+
+Usage:
+    python src/scripts/push_coral_to_hub.py\
+            <path/to/recording/metadata/path> <path/to/speaker/metadata/path> <hub_id>
+"""
+
+from pathlib import Path
+from datasets import Dataset, Audio, DatasetDict
+from datetime import datetime
+import pandas as pd
+import click
+import logging
+import time
+
+from requests import HTTPError
+
+logging.basicConfig(level=logging.INFO, format="%(asctime)s | %(message)s")
+logger = logging.getLogger(__name__)
+
+test_speaker_ids = [
+    "t16023910",
+    "t22996947",
+    "t17053799",
+    "t47310812",
+    "t59821643",
+    "t79384144",
+    "t37263163",
+    "t39126337",
+    "t22996947",
+    "t17419826",
+    "t29982093",
+    "t40224720",
+    "t27567090",
+    "t31776488",
+    "t40353825",
+    "t34653502",
+    "t26322580",
+    "t72771561",
+    "t48392154",
+    "t38841910",
+    "t36170006",
+    "t10062436",
+    "t21820268",
+    "t39414039",
+    "t13282221",
+    "t82923090",
+    "t35107011",
+    "t13330719",
+    "t33840200",
+    "t10367179",
+    "t39656524",
+    "t37619007",
+    "t42345465",
+    "t21902156",
+]
+
+
+@click.command("Builds and pushes the CoRal test dataset.")
+@click.argument("recording_metadata_path", type=click.Path(exists=True))
+@click.argument("speaker_metadata_path", type=click.Path(exists=True))
+@click.argument("hub_id", type=str)
+@click.argument("major_version", type=int, default=1, show_default=True)
+@click.argument("minor_version", type=int, default=0, show_default=True)
+@click.option(
+    "--private",
+    is_flag=True,
+    default=True,
+    show_default=True,
+    help="Whether to make the dataset private on the Hugging Face Hub.",
+)
+def main(
+    recording_metadata_path: Path | str = Path("data/processed/recordings.xlsx"),
+    speaker_metadata_path: Path | str = Path("data/hidden/speakers.xlsx"),
+    hub_id: str = "alexandrainst/coral",
+    major_version: int = 1,
+    minor_version: int = 0,
+    private: bool = True,
+) -> None:
+
+    # Load the metadata and split into test/train speakers
+    recording_metadata_path = Path(recording_metadata_path)
+    recording_metadata = pd.read_excel(recording_metadata_path, index_col=0)
+
+    # Change the dtype of all columns to string
+    recording_metadata = recording_metadata.astype(str)
+
+    # All recordings which were made before 2024-03-11 are defined as iteration 1
```

Comment:
This is not really a great way to indicate the start/end of different iterations -  any suggestions? :)